modules/network/src/main/java/org/apache/ignite/internal/network/serialization/ClassDescriptor.java

```diff
@@ -59,6 +65,20 @@
      */
     private final boolean isFinal;
 
+    /** Total number of bytes needed to store all primitive fields. */
+    private final int primitiveFieldsDataSize;
+    /** Total number of object (i.e. non-primitive) fields. */
+    private final int objectFieldsCount;
+
+    private Map<String, FieldDescriptor> fieldsByName;
+    /**
+     * Offsets into primitive fields data (which has size {@link #primitiveFieldsDataSize}).
+     * These are different from the offsets used in the context of {@link sun.misc.Unsafe}.
```

Comment:
Why are they different? Needs elaboration

Reply:
'primitive fields data' is a byte array to which we collect values of all primitive fields of the object. 'Offsets' are indices in this array. Unsafe has nothing to do with this.

Do you suggest elaborating this in the comment even more?

Reply:
Yes, I think array should be mentioned because it's not clear what are these offsets

Reply:
Mentioned the array and what it is